WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEmailFragment.java

```diff
@@ -460,10 +462,14 @@ public void onSocialChanged(OnSocialChanged event) {
                     event.error.type.toString(), event.error.message);
 
             switch (event.error.type) {
+                // WordPress account does not exist with input email address.
                 case UNKNOWN_USER:
                     showErrorDialog(getString(R.string.login_error_email_not_found, mGoogleEmail));
                     break;
+                // Unknown error.
+                case GENERIC_ERROR:
```

Comment:
The GENERIC_ERROR falls through to the default case so, is there a particular reason to have a dedicated case for it? If yes, but wanting it empty, let's have a comment here to explain the fall through. Thanks!

In the end, it seems that we're only interested in differentiating the UNKNOWN_USER case so, a simple if statement or a ternary operator would suffice. In general, I prefer not to use default unless it's really important to do that shortcircuiting.

Reply:
The `GENERIC_ERROR` case is included to explicitly show all error types and the `default` case is to cover anything returned from the API that is unexpected.  I added a comment in e819d40c3f617b45e0670adb28deb788abc436b5.

More cases are added in a subsequent pull request which is why a switch statement is used rather than a ternary operator.

Reply:
The `GENERIC_ERROR` case is included to explicitly show all error types and the `default` case is to cover anything returned from the API that is unexpected.  I added a comment in e819d40c3f617b45e0670adb28deb788abc436b5.

More cases are added in a subsequent pull request which is why a switch statement is used rather than a ternary operator.

Reply:
Took the liberty to add commit to make it clear that "Do nothing" [here](https://github.com/wordpress-mobile/WordPress-Android/commit/e819d40c3f617b45e0670adb28deb788abc436b5) means fall through since the case does open up an error dialog after all.